tests/plotting/test_plotting.py

```diff
@@ -521,11 +521,31 @@ def test_plot_bounds_axes_with_no_data():
 
 def test_plot_show_grid(sphere):
     plotter = pyvista.Plotter()
+
+    with pytest.raises(ValueError, match='Value of location'):
+        plotter.show_grid(location='foo')
+    with pytest.raises(TypeError, match='location must be a string'):
+        plotter.show_grid(location=10)
+    with pytest.raises(ValueError, match='Value of ticks'):
+        plotter.show_grid(ticks='foo')
+    with pytest.raises(TypeError, match='ticks must be a string'):
+        plotter.show_grid(ticks=10)
+
     plotter.show_grid()
     plotter.add_mesh(sphere)
     plotter.show(before_close_callback=verify_cache_image)
 
 
+def test_plot_show_grid_with_mesh(hexbeam, plane):
+    """Show the grid bounds for a specific mesh."""
+    hexbeam.clear_data()
+    plotter = pyvista.Plotter()
+    plotter.add_mesh(hexbeam, style='wireframe')
+    plotter.add_mesh(plane)
+    plotter.show_grid(mesh=plane)
+    plotter.show(before_close_callback=verify_cache_image)
```

Comment:
What's up with that `Z axis (x10^-18)` annotation? Is it applying the bounds to the plane, which being 2d has 0 Z range? Should we switch the order of the plane and the beam being added just to make this a bit nicer?

Reply:
Might be better to just disable showing the z axis labels. If this was for an example I’d agree with changing it, but if it’s just for a unit test, there’s not much of a point in making it look nice.

Reply:
It might also happen that the `1e-18` noise ends up changing between systems or VTK versions, but disabling it altogether will prevent that too.

Reply:
Good point. I’ll disable it.

Reply:
You could also set the label format to be not in scientific format.

Reply:
Done in 366e4f67152d08735e46748f81eecc4b634fd1.